src/graph/connect.cc

```diff
@@ -435,6 +435,50 @@ static int copyChannels(struct ncclComm* comm, int start, int end, int* ringPrev
   return c;
 }
 
+static ncclResult_t Rewriteinter(struct ncclTopoSystem* system,struct ncclTopoGraph* graph,int nChannels){
+  int nnets = system->nodes[NET].count;
+  int ngpus = system->nodes[GPU].count;
+  typedef struct {
+    int id;
+    int used;
+    } Net;
+  Net nets[nnets];
+  memset(nets, 0, nnets*sizeof(Net));
+  for (int i = 0; i < nnets; i++) {
+    nets[i].id = system->nodes[NET].nodes[i].id;
+  }
+  for (int i = 0; i < nChannels; i++){
+    memcpy(graph->intra+i*ngpus, graph->intra, ngpus*sizeof(int));
+    qsort(nets, nnets, sizeof(Net), [](const void* a, const void* b)->int {
+    return ((Net*)a)->used - ((Net*)b)->used;
+    });
+    *(graph->inter+i*2) = nets[0].id;
```

Comment:
Why `2` here?

Reply:
Add comments for the magic values and why you are sorting everytime a NIC is used?

Reply:
Since graph->inter[] means the list of NICs,"i" indicates the channel ID,graph->inter[i*2] indicates the NIC where the channel i enters the node,graph->inter[i*2 +1] indicates the NIC where the channel i leaves the node.
I am sorting everytime a NIC is used because I need to make sure we can use NICs averagely.